papyri/gen.py

```diff
@@ -550,6 +551,8 @@ def gen_main(
 
     if not target_dir.exists() and not config.dry_run:
         target_dir.mkdir(parents=True, exist_ok=True)
+    if dry_run:
+        target_dir = tempfile.mkdtemp()
```

Comment:
now that we have that, we should be able to remove the:

`if not config.dry_run:` can be removed on line 573, it can be unconditional as now we have a temporary directory.
The goal being to test that the writing logic actually works.

I'm also wondering if we coudl use `tempfile.TemporaryDirectory` as a contet manager, but that my be more complicated.

And instead of using  tempfile.mkdtemp()